src/components/PremiumTodos.js

```diff
@@ -1,38 +1,65 @@
 import React, { useState, useEffect } from 'react';
 import { filterTodos, sortTodos } from '../helpers/editTodos';
-import { TodosInput }  from './TodoInput';
+import { TodosInput } from './TodoInput';
 import { Todos } from './Todo';
 import { DataStore } from 'aws-amplify';
 import { Todo } from '../models';
-import { ShowTodosDone } from './Button';
+import { Button, ShowTodosDone } from './Button';
 import '../styles.css';
+import { Link } from 'react-router-dom';
+import { useAuthenticator } from '@aws-amplify/ui-react';
+
+export function PremiumTodos() {
+    const { user, signOut } = useAuthenticator((context) => [context.user]);
 
-export function PremiumTodos (){
     const [listTodos, setListTodos] = useState([]);
     const [showTodosDone, setShowTodosDone] = useState(null);
-
     useEffect(() => {
-        const subscription = DataStore.observeQuery(Todo).subscribe(msg => {
+        DataStore.observeQuery(Todo).subscribe(msg => {
             setListTodos(msg.items);
-          });
-
-    },[]);
+        });
+    }, []);
 
-    const displayTodosDone = ()=> {
+    const displayTodosDone = () => {
         setShowTodosDone(!showTodosDone)
     }
-   
+
     return (
         <div className='card'>
             <div className='header'>Todoco</div>
-                <TodosInput listTodos={listTodos}/>
-                <Todos listTodos={sortTodos(listTodos.filter((todo)=> !todo.done ))}/>
-                <div className='listTodosDone' onClick={displayTodosDone}>Completed ({filterTodos(listTodos).length})
-                    {filterTodos(listTodos).length !== 0 ? <ShowTodosDone className='iconShowTodosDone' d={showTodosDone ? "M3 4.5h14.25M3 9h9.75M3 13.5h5.25m5.25-.75L17.25 9m0 0L21 12.75M17.25 9v12" :
-                        "M3 4.5h14.25M3 9h9.75M3 13.5h9.75m4.5-4.5v12m0 0l-3.75-3.75M17.25 21L21 17.25"}
-                        onClick={displayTodosDone}/> : ""}
+            <TodosInput listTodos={listTodos} />
+            <Todos listTodos={sortTodos(listTodos.filter((todo) => !todo.done))} />
+            <div className='listTodosDone' onClick={displayTodosDone}>Completed ({filterTodos(listTodos).length})
```

Comment:
We should create variables to all this function calls.
```js
const sortedTodos = sortTodos(listTodos.filter((todo) => !todo.done));
const doneTodos = filterTodos(listTodos); // better name: filterDoneTodos
```
Also, `filterTodos` can be named better since it does not say what is filtering. Maybe `filterDoneTodos`.